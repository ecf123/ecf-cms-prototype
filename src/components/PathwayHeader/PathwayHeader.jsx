import "./PathwayHeader.scss"
const PathwayHeader = () => {
  return (
    <div>PathwayHeader</div>
  )
}

export default PathwayHeader