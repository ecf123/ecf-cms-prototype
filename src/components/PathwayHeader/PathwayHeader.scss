@use "../../styles/_typography" as *;
@use "../../styles/_variables.scss" as *;

.heading {
  display: flex;
  flex-direction: column;
  width: 46.8rem;
  gap: 1.25rem;

  &__image {
    position: relative;
    width: 100%;
    height: auto;
  }

  &__icon {
    position: absolute;
    z-index: 1;
    width: 5.6rem;
    height: 5.6rem;
    top: 1.25rem;
    left: 1.25rem;
  }

  &__title {
    color: $color-blue-navy-dark;
  }
}
