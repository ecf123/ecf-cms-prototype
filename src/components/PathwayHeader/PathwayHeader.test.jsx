import PathwayHeader from "./PathwayHeader";
import { customRender } from "../../utils/testUtils";
import { render, screen } from "@testing-library/react";
import headingImage from "../../assets/mobile-growing-investment.svg"
import companyIcon from "../../assets/barclays-icon-white-background.svg"

it("should render the pathway header with props", () => {
    const { container } = customRender(
        <PathwayHeader headingTitle = "Commercial Banking with Barclay's" 
        headingImage = {headingImage} 
        companyIcon = {companyIcon} 
        links = {["Finance", "Investing"]} 
        condition = {false}/>); 
    expect(container).toMatchSnapshot();
});

it("should render the pathway header title", () => {
    render(<PathwayHeader headingTitle = "Commercial Banking with Barclay's" 
        headingImage = {headingImage} 
        companyIcon = {companyIcon} 
        links = {["Finance", "Investing"]} 
        condition = {false}/>);
    const title = screen.getByRole("heading");
    expect(title).toBeInTheDocument();
  });
  
it("should render the pathway header links", () => {
    render(<PathwayHeader headingTitle = "Commercial Banking with Barclay's" 
        headingImage = {headingImage} 
        companyIcon = {companyIcon} 
        links = {["Finance", "Investing"]} 
        condition = {false}/>);
    const linksOne = screen.getByText(/Finance/);
    const linksTwo = screen.getByText(/Investing/);
    expect(linksOne).toBeInTheDocument();
    expect(linksTwo).toBeInTheDocument();
});

it("should render the pathway header image and icon", () => {
    render(<PathwayHeader headingTitle = "Commercial Banking with Barclay's" 
        headingImage = {headingImage} 
        companyIcon = {companyIcon} 
        links = {["Finance", "Investing"]} 
        condition = {false}/>);
    const image = screen.getByAltText("Company Icon");
    const icon = screen.getByAltText("Pathway Heading");
    expect(image).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
});
