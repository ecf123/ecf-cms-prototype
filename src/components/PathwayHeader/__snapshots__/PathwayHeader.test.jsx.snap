// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the pathway header with props 1`] = `
<div>
  <div
    class="heading"
  >
    <img
      alt="Barclays Icon"
      class="heading__icon"
      src="barclays-icon-white-background.svg"
    />
    <img
      alt="Mobile Growing Investment"
      class="heading__image"
      src="mobile-growing-investment.svg"
    />
    <h1
      class="heading__title"
    >
      Commercial Banking with Barclay's
    </h1>
    <ul
      class="categories-menu"
    >
      <li
        class="categories-menu__item categories-menu__item--purple"
      >
        Finance
      </li>
      <li
        class="categories-menu__item categories-menu__item--darkpurple"
      >
        Investing
      </li>
    </ul>
  </div>
</div>
`;
